includes/class-wc-stripe-helper.php

```diff
@@ -722,6 +722,52 @@ public static function get_upe_settings_enabled_payment_method_ids( $gateway ) {
 		return $enabled_gateways;
 	}
 
+	/**
+	 * Reorders the list of available payment gateways in 'woocommerce_gateway_order' option to include the Stripe methods
+	 * in the order merchants have chosen in the settings.
+	 *
+	 * @param array $ordered_payment_method_ids Ordered Stripe payment method list.
+	 */
+	public static function add_stripe_methods_in_woocommerce_gateway_order( $ordered_payment_method_ids ) {
+		$gateway_order = get_option( 'woocommerce_gateway_order' );
+
+		$ordered_available_stripe_methods = [];
+		// Map the Stripe payment method list to the right format to save in the 'woocommerce_gateway_order' option.
+		foreach ( $ordered_payment_method_ids as $payment_method_id ) {
+			if ( 'card' === $payment_method_id ) {
+				$gateway_id                         = 'stripe';
+				$ordered_available_stripe_methods[] = $gateway_id;
+				continue;
+			} elseif ( 'sepa_debit' === $payment_method_id ) {
+				$gateway_id = 'stripe_sepa';
+			}
+
+			$gateway_id = str_starts_with( $payment_method_id, 'stripe_' ) ? $payment_method_id : 'stripe_' . $payment_method_id;
```

Comment:
> Is this check for whether $payment_method_id starts with stripe_ used for legacy? Asking because I don't see any $payment_method_id to start with stripe_, but they do on legacy.

This check is for new experience (UPE). In legacy the method names already have `stripe_` but not in UPE. The `woocommerce_gateway_order` option is supposed to keep the gateway names with `stripe_`.

> If so, would it be better to use the array passed to this method before we [prepend](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3222/files#diff-679186f0c3e522393b5394f6e65890f1b2bff5ef64efb250eee90d6348dc841eR316) stripe_ to the array values?

We are also calling this function [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3222/files#diff-4c2be2e009983410faf0c92dfb7770331510dfc470e5c156d5055fbf46f324f5R56) from `set_stripe_gateways_in_list` function. If we do the mapping before passing to the function, we will have to do it twice.